pyproject.toml

```diff
@@ -25,7 +25,11 @@ dependencies = [
     "numpy>=2.0.0",
     "packaging",
     "universal-pathlib",
+    "h5py",
+    "hdf5plugin",
     "numcodecs",
+    "imagecodecs",
+    "imagecodecs-numcodecs==2024.6.1",
```

Comment:
I think these should be optional no? (`numcodecs` is still required). That also means they can be removed from the `min-deps.yml` in CI.